test/test_linearizer.py

```diff
@@ -270,31 +274,39 @@ def test_double_reduce_multireduce(self):
     ]
     lins = helper_linearizer_ast(sink, [x], wanna_output=[wanna_output], opts=opts)
     for l in lins:
+      if Device.DEFAULT not in {"METAL"}:
+        assert sum(1 if u.op is UOps.SPECIAL else 0 for u in l.uops)-l.global_dims-l.local_dims <= 2, "shouldn't be more than 2 explicit local dims"
       ranges = [u.op for u in l.uops if (u.op is UOps.RANGE and u.arg[1]) or (u.op is UOps.ENDRANGE and u.src[0].arg[1])]
       for i,u in enumerate(ranges):
         if i < 2: continue
         assert ranges[i-2] != u or ranges[i-1] != u, f"multireduce nested the ranges! {ranges[i-2], ranges[i-1], {u}}"
+    # check for correctness when dims are grouped differently
+    helper_linearizer_ast(sink, [x], wanna_output=[wanna_output], \
+      opts=[[Opt(OptOps.GROUPTOP, 0, 4), Opt(OptOps.GROUPTOP, 1, 2), Opt(OptOps.GROUPTOP, 2, 4), Opt(OptOps.GROUPTOP, 3, 8)]])
 
   @unittest.skipIf(CI and Device.DEFAULT in {"AMD"}, "AMD CI doesn't support multiple sync threads yet")
   @unittest.skipUnless(Device[Device.DEFAULT].renderer.has_local, "test requires locals")
   @unittest.skipUnless(Device[Device.DEFAULT].renderer.has_shared, "test requires shared")
   def test_partial_opt_multireduce(self):
     # check how it works with one reduce optimized and one unoptimized
     Tensor.manual_seed(0)
-    x = Tensor.randn(27, 15, 5, dtype=dtypes.float).softmax(1).realize()
+    x = Tensor.randn(27, 12, 5, dtype=dtypes.float).softmax(1).realize()
```

Comment:
why did this change?

Reply:
I was playing with different combinations of divisors, I can change it back